misc/services/mobile-webui/mobile-webui-frontend/src/reducers/wfProcesses_status/picking.js

```diff
@@ -148,12 +284,89 @@ const normalizePickingLines = (lines) => {
+        //     huBarcode: '1000437',
+        //     uom: 'Kg',
+        //     qtyAvailable: 45,
+        //     qtyPicked: 0,
+        //   },
+        //   1000820: {
+        //     id: '1000820',
+        //     locatorName: 'Hauptlager',
+        //     huBarcode: '1000463',
+        //     uom: 'Kg',
+        //     qtyAvailable: 25,
+        //     qtyPicked: 0,
+        //   },
+        // };
+
         return accum;
       }, {}),
     };
   });
 };
 
+export const mergeActivityDataStoredAndGenerateAltSteps = ({ draftActivityDataStored, fromActivity }) => {
+  const { lines } = fromActivity.componentProps;
+  let genSteps = {};
+
+  // loop within steps
+  for (let lineIdx = 0; lineIdx < lines.length; lineIdx++) {
+    // console.log('ABOUT_TO_SEND:', draftActivityDataStored.dataStored.lines[lineIdx]);
+    computeLineStatus({ draftLine: draftActivityDataStored.dataStored.lines[lineIdx] });
+
+    for (let stepIdx = 0; stepIdx < lines[lineIdx].steps.length; stepIdx++) {
+      let step = lines[lineIdx].steps[stepIdx];
+      let { qtyRejected } = lines[lineIdx].steps[stepIdx].mainPickFrom;
+      let totalAltQtys = 0;
+
+      // allocateQtys in sync with the data received from the BE
+      let pickFromAlternatives = fromActivity.componentProps.pickFromAlternatives;
+      for (let altKey in pickFromAlternatives) {
+        if (
+          step.pickFromAlternatives[pickFromAlternatives[altKey].id] &&
+          step.pickFromAlternatives[pickFromAlternatives[altKey].id].qtyPicked > 0
+        ) {
+          totalAltQtys = totalAltQtys + step.pickFromAlternatives[pickFromAlternatives[altKey].id].qtyPicked;
+
+          genSteps[pickFromAlternatives[altKey].id] = pickFromAlternatives[altKey];
+          genSteps[pickFromAlternatives[altKey].id].qtyPicked =
+            step.pickFromAlternatives[pickFromAlternatives[altKey].id].qtyPicked;
+          pickFromAlternatives[altKey].allocatedQtys = {
+            [pickFromAlternatives[altKey].id]: pickFromAlternatives[altKey].qtyAvailable,
+          };
+        } else {
+          pickFromAlternatives[altKey].allocatedQtys = {};
+        }
+      }
+      draftActivityDataStored.dataStored.lines[lineIdx].steps[step.pickingStepId].pickFromAlternatives =
+        pickFromAlternatives;
+
+      draftActivityDataStored.dataStored.lines[lineIdx].steps[step.pickingStepId].altSteps.genSteps = genSteps;
+
+      // In case we have no generated steps and there is a qtyRejected to be filled - we need to generate those alternatives
+      const remainingQtyDiff = qtyRejected - totalAltQtys;
+
+      if ((Object.keys(genSteps).length === 0 && qtyRejected) || remainingQtyDiff > 0) {
+        draftActivityDataStored.dataStored = generateAlternativeSteps({
+          draftDataStored: draftActivityDataStored.dataStored,
+          lineId: lineIdx,
+          stepId: step.pickingStepId,
+          qtyToAllocate: qtyRejected,
+        });
+      }
+    }
+  }
+
+  return draftActivityDataStored;
```

Comment:
you are returning exactly the same thing you get as param.
why is needed to return it?